mungegithub/Makefile

```diff
@@ -22,6 +22,13 @@ hooksecret64=$(shell base64 $(HOOKSECRET))
 
 READONLY ?= true
 
+CLUSTER ?= mungegithub
+PROJECT ?= k8s-mungegithub
+ZONE ?= us-central1-b
+
+get-cluster-credentials:
+	gcloud container clusters get-credentials "$(CLUSTER)" --project="$(PROJECT)" --zone="$(ZONE)"
```

Comment:
this does not belong into a Makefile of a generic infrastructure project, does it?

Reply:
We use this pattern in Prow as well:
https://github.com/kubernetes/test-infra/blob/566f78c17e7761be7deb888b0b080cb7f1dfe3ce/prow/Makefile#L46-L47

The `get-cluster-credentials` target is useful to avoid deploying to the wrong cluster.
As for defaulting to the clusters that run the kubernetes instances of these projects, I think it is worth keeping these defaults even if it makes the Makefiles less general because an oncall developer won't have to look up the cluster credentials in an emergency.

The Makefiles can still be used with other clusters by setting the `CLUSTER`, `PROJECT` and `ZONE` env vars, they just default to using the clusters that run the kubernetes deployments of these projects.

Reply:
But there is a world outside of gcloud.

Reply:
I don't think having this target hurts as long as it is documented, but maybe the other targets shouldn't depend on it.

This is a common task that we need to do a lot while maintaining Kubernetes' deployments and even once we have CNCF projects those with access will still need a quick way to switch to the right cluster without needing to keep up with those 3 settings.

[Mungegithub is essentially deprecated anyhow](https://github.com/kubernetes/test-infra/issues/3331)

Reply:
If it is not mandatory, it's fine. Was falling over it as the staging repo publisher runs in OpenShift for development.

Reply:
I'll put out a PR to fix that :smile:

Reply:
Oh hmm, let's open an issue to discuss actually: https://github.com/kubernetes/test-infra/issues/4814